src/Common/ThreadStatus.h

```diff
@@ -204,17 +221,11 @@ class ThreadStatus : public boost::noncopyable
     {
         DetachedFromQuery = 0,  /// We just created thread or it is a background thread
         AttachedToQuery,        /// Thread executes enqueued query
-        Died,                   /// Thread does not exist
     };
 
-    int getCurrentState() const
```

Comment:
Thank you for removing this. There is no reason why `memory_order_relaxed` was used